tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py

```diff
@@ -701,8 +701,9 @@ def test_create_model_version_gcp(store, local_model_dir, create_args):
             notebook_entity = Notebook(id=str(notebook_id))
             entity = Entity(notebook=notebook_entity)
             lineage_header_info = LineageHeaderInfo(entities=[entity])
+            expected_lineage_header = json.dumps(json.loads(message_to_json(lineage_header_info)))
```

Comment:
Similar to https://github.com/mlflow/mlflow/pull/8638/files#r1220837005, can we add a comment explaining the importance of this assert? Can we also assert that the `expected_lineage_header` doesn't contain any newlines?